test/suite/test_alter03.py

```diff
@@ -0,0 +1,81 @@
+#!/usr/bin/env python
+#
+# Public Domain 2014-2018 MongoDB, Inc.
+# Public Domain 2008-2014 WiredTiger, Inc.
+#
+# This is free and unencumbered software released into the public domain.
+#
+# Anyone is free to copy, modify, publish, use, compile, sell, or
+# distribute this software, either in source code form or as a compiled
+# binary, for any purpose, commercial or non-commercial, and by any
+# means.
+#
+# In jurisdictions that recognize copyright laws, the author or authors
+# of this software dedicate any and all copyright interest in the
+# software to the public domain. We make this dedication for the benefit
+# of the public at large and to the detriment of our heirs and
+# successors. We intend this dedication to be an overt act of
+# relinquishment in perpetuity of all present and future rights to this
+# software under copyright law.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+# EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+# MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+# IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+
+import wiredtiger, wttest
+from wtscenario import make_scenarios
+
+# test_alter03.py
+#    Check if app_metadata can be altered.
+class test_alter03(wttest.WiredTigerTestCase):
+    name = "alter03"
+
+    def verify_metadata(self, metastr):
+        if metastr == '':
+            return
+        cursor = self.session.open_cursor('metadata:', None, None)
+        #
+        # Walk through all the metadata looking for the entries that are
```

Comment:
This function is kind of common between all alter tests. I mostly just copied it.

We expect to find multiple URIs that match the metastr, hence we do not break out of the loop. The comment here says "... looking for the entries that are the URIs for the named object." I think once you understand that there will be multiple objects matching the name (because, for instance, a table has multiple objects) it is obvious that we want to check them all, and not break out of the loop.